pkg/api/addons_test.go

```diff
@@ -1564,25 +1564,33 @@ func TestSetAddonsConfig(t *testing.T) {
 					Enabled: to.BoolPtr(true),
 					Config: map[string]string{
 						"omsAgentVersion":       "1.10.0.1",
-						"dockerProviderVersion": "10.0.0-1",
+						"dockerProviderVersion": "12.0.0-0",
 						"schema-versions":       "v1",
 						"clusterName":           "aks-engine-cluster",
 						"workspaceDomain":       "b3BpbnNpZ2h0cy5henVyZS5jb20=",
 					},
 					Containers: []KubernetesContainerSpec{
 						{
 							Name:           "omsagent",
+							CPURequests:    "75m",
+							MemoryRequests: "225Mi",
+							CPULimits:      "500m",
+							MemoryLimits:   "600Mi",
+							Image:          "mcr.microsoft.com/azuremonitor/containerinsights/ciprod:ciprod01112021",
+						},
+						{
+							Name:           "omsagent-rs",
 							CPURequests:    "150m",
 							MemoryRequests: "250Mi",
 							CPULimits:      "1",
-							MemoryLimits:   "750Mi",
-							Image:          "mcr.microsoft.com/azuremonitor/containerinsights/ciprod:ciprod10052020",
+							MemoryLimits:   "1Gi",
```

Comment:
This change wants to increase the request-limit ratio from 3x to 4x. The recommended ratio is 2x, to eliminate as much memory reservation variability in practice, in order to improve deterministic pod scheduling (and optimize pod density across the cluster).

Why do we want to continue to increase the amount of memory that is available? Is this application particularly "spiky"?

Reply:
sorry, this comment should have been moved up to the actual implemntation, not the UT :)

Reply:
Currently, our replica set is singleton and for large workloads, say high number of pods, events and nodes, or high prometheus scraping requires higher memory limit.  And also we have seen if the node where  replicaset got sheduled has disk I/O issues and the high memory usage. We have optimized our disk I/O paths to avoid higher memory usage and other perf improvements in this release. Currently, We are working on moving prometheus scraping to deamonset so that load will be evenly distributed and moved way from singleton replicaset.   Memory requests and limits what we have here is inline what we have default in AKS and also accommodate these defaults for large clusters.